c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1439 @@
+//! \{
+
+//! Timer control register
+typedef struct {
+    uint one_shot : 1;          //!< 0 = wrapping mode, 1 = one shot
+    uint size : 1;              //!< 0 = 16 bit, 1 = 32 bit
+    uint pre_divide : 2;        //!< divide input clock (see ::timer_pre_divide)
+    uint : 1;
+    uint interrupt_enable : 1;  //!< enable interrupt (1 = enabled)
+    uint periodic_mode : 1;     //!< 0 = free-running; 1 = periodic
+    uint enable : 1;            //!< enable counter/timer (1 = enabled)
+    uint : 24;
+} timer_control_t;
+
+//! Values for ::timer_control_t::pre_divide
+enum timer_pre_divide {
+    TIMER_PRE_DIVIDE_1 = 0,     //!< Divide by 1
+    TIMER_PRE_DIVIDE_16 = 1,    //!< Divide by 16
+    TIMER_PRE_DIVIDE_256 = 2    //!< Divide by 256
+};
+
+//! Timer interrupt status flag
+typedef struct {
+    uint status : 1;            //!< The flag bit
+    uint : 31;
+} timer_interrupt_status_t;
+
+//! Timer controller registers
+typedef struct {
+    uint load_value;            //!< Load value for Timer
+    const uint current_value;   //!< Current value of Timer
+    timer_control_t control;    //!< Timer control register
+    uint interrupt_clear;       //!< Interrupt clear (any value may be written)
+    //! Timer raw interrupt status
+    const timer_interrupt_status_t raw_interrupt_status;
+    //! Timer masked interrupt status
+    const timer_interrupt_status_t masked_interrupt_status;
+    uint background_load_value; //!< Background load value for Timer
+    uint _dummy;
+} timer_controller_t;
+
+ASSERT_WORD_SIZED(timer_control_t);
+ASSERT_WORD_SIZED(timer_interrupt_status_t);
+
+//! Timer 1 control registers
+static volatile timer_controller_t *const timer1_control =
+        (timer_controller_t *) TIMER1_BASE;
+//! Timer 2 control registers
+static volatile timer_controller_t *const timer2_control =
+        (timer_controller_t *) TIMER2_BASE;
+
+//! \}
+
+// ---------------------------------------------------------------------
+//! \name 7. DMA Controller
+//! \brief Each ARM968 processing subsystem includes a DMA controller.
+//! \details The DMA controller is primarily used for transferring inter-neural
+//!     connection data from the SDRAM in large blocks in response to an input
+//!     event arriving at a fascicle processor, and for returning updated
+//!     connection data during learning. In addition, the DMA controller can
+//!     transfer data to/from other targets on the System NoC such as the
+//!     System RAM and Boot ROM.
+//!
+//!     As a secondary function the DMA controller incorporates a ‘Bridge’
+//!     across which its host ARM968 has direct read and write access to
+//!     System NoC devices, including the SDRAM. The ARM968 can use the
+//!     Bridge whether or not DMA transfers are active.
+//! \{
+
+//! DMA descriptor
+typedef struct {
+    uint _zeroes : 2;       //!< Must be zero
+    uint length_words : 15; //!< length of the DMA transfer, in words
```

Comment:
named constant which can then have a comment explaing what the 15 words are.

Reply:
No.